### PR TITLE
[TRA-416] Set market mapper revenue share on market creation

### PR DESCRIPTION
**Pull request by shrenujb**
### Test Plan
Added tests

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
  - Integration with the new RevShare module for improved revenue sharing capabilities.

- **Bug Fixes**
  - Enhanced test setups to properly initialize all required dependencies for accurate test results.

- **Tests**
  - Updated tests to include additional parameters ensuring comprehensive test coverage.

- **Chores**
  - Adjusted various initialization functions to include new dependencies for better modularity and code maintainability.

### CODE REVIEW

File: protocol/x/prices/keeper/market_test.go [hunk 1]
```diff
@@ -57,6 +57,11 @@ func TestCreateMarket(t *testing.T) {
 
 	// Verify expected market event.
 	keepertest.AssertMarketCreateEventInIndexerBlock(t, keeper, ctx, marketParam)
+
+	// Verify market revenue share creation
+	revShareDetails, err := revShareKeeper.GetMarketMapperRevShareDetails(ctx, marketParam.Id)
+	require.NoError(t, err)
+	require.True(t, revShareDetails.ExpirationTs > 0)
```
Comment (vincentwschau) [inline protocol/x/prices/keeper/market_test.go:64]: nit: Can you verify the value is some specific number rather than just > 0?

Comment (shrenujb) [reply protocol/x/prices/keeper/market_test.go:64]: The value will be set based on ctx.Blocktime which will be variable so I can't test for that.
Unless there's a way to get a deterministic time?

Comment (vincentwschau) [reply protocol/x/prices/keeper/market_test.go:64]: Can't you get ctx.Blocktime and check against some variable that is blocktime + valid days?

File: protocol/app/app.go [hunk 2]
```diff
@@ -890,19 +890,34 @@ func New(
...
+	)
+	revShareModule := revsharemodule.NewAppModule(appCodec, app.RevShareKeeper)
+
 	app.PricesKeeper = *pricesmodulekeeper.NewKeeper(
 		appCodec,
 		keys[pricesmoduletypes.StoreKey],
 		indexPriceCache,
 		timeProvider,
 		app.IndexerEventManager,
-		// set the governance and delaymsg module accounts as the authority for conducting upgrades
```
Comment (vincentwschau) [inline protocol/app/app.go:899]: nit: Why is this comment removed?

**Review by vincentwschau — APPROVED** [review]

**Pull request merged**